islandora_video_streaming_panopto.module

```diff
@@ -40,20 +57,256 @@ function islandora_video_streaming_panopto_islandora_sp_videoCModel_islandora_de
+ */
+function islandora_video_streaming_panopto_islandora_datastream_alter(AbstractObject $object, AbstractDatastream $datastream, &$context) {
+  if ($datastream->id == 'PANOPTO_INFO') {
+    islandora_video_streaming_panopto_delete_from_panopto($object, $context);
+  }
+}
+
+/*
+ * Deletes video from Panopto server.
+ * @param AbstractObject $object The object being altered
+ * @param array $context The context or action of which the object will be altered
+ */
+function islandora_video_streaming_panopto_delete_from_panopto(AbstractObject $object, &$context) {
+  if (isset($object['PANOPTO_INFO']) && ($context['action'] == 'purge' || $context['action'] == 'delete' )) {
+   //Panopto/api/v1/sessions/{id} for deleting videos on Panopto
+    $decoded = json_decode($object['PANOPTO_INFO']->content);
+    if (isset($decoded)) {
+      $id = $decoded->upload_id;
+      if (isset($id)) {
+        $oauthendpoint = variable_get('islandora_video_streaming_panopto_host');
+        $uploadendpoint = $oauthendpoint . '/Panopto/api/v1/sessions/'. $id;
+        $oauthendpoint .= "/Panopto/oauth2/connect/token";
+        $token = islandora_video_streaming_panopto_get_token($oauthendpoint);
+        if (isset($token)) {
+          $headers = array("Authorization: Bearer {$token}", "Content-Type: application/json");
+          $response = islandora_video_streaming_panopto_request_curl($uploadendpoint, $headers, '', 'DELETE');
+        }
+      }
     }
   }
 }
 
+/**
+ * Requests token and returns token
+ * @param string $oauthendpoint The authentication URL for request
+ * @return string|null A token retrieved from server, or null on failure
+ */
+function islandora_video_streaming_panopto_get_token($oauthendpoint) {
+  $client_id = variable_get('islandora_video_streaming_panopto_client_id');
+  $secret_client_id = variable_get('islandora_video_streaming_panopto_secret_client_id');
+  $username = variable_get('islandora_video_streaming_panopto_username');
+  $password = variable_get('islandora_video_streaming_panopto_password');
+  
+  $response = null;
+  $token = null;
+  
+  if (isset($oauthendpoint) && isset($client_id) && isset($secret_client_id) && isset($username) && isset($password)) {
+    $authorization = base64_encode("$client_id:$secret_client_id");
+    $headers = array("Authorization: Basic {$authorization}", "Content-Type: application/x-www-form-urlencoded");
+    $content = 'grant_type=password&username='.urlencode(strtolower($username)).'&password='.urlencode($password).'&scope=api';
+    $response = islandora_video_streaming_panopto_request_curl($oauthendpoint, $headers, $content, 'POST', array(200 => 'access_token'));
+  }
+  if ($response) {
+    $token = $response['access_token'];
+  } else {
+    $messages = array(
+      'message' => t('Failed to retrieve $token. $response is @response . $header is @header . $content is @content'),
+      'message_sub' => array(
+        '@response' => $response,
+        '@header' => $header,
```

Comment:
`$header` appears to be a typo of `$headers`.  Note also that this would log the client id and secret to the watchdog, which is sensitive and probably not helpful in debugging.

Reply:
Taking out the $header would be best then. I will do the same to the other $headers.

Reply:
`$content` here similarly contains the username / password, and is probably low value for debugging.